drivers/sensor/bq274xx/bq274xx.c

```diff
@@ -636,26 +637,160 @@ static int bq274xx_gauge_configure(const struct device *dev)
 		return -EIO;
 	}
 
+#ifdef CONFIG_PM_DEVICE
+	bq274xx->pm_state = DEVICE_PM_ACTIVE_STATE;
+#endif
+
+	return 0;
+}
+
+#ifdef CONFIG_PM_DEVICE
+static int bq274xx_enter_shutdown_mode(struct bq274xx_data *data)
+{
+	int status = 0;
+
+	status = bq274xx_control_reg_write(data, BQ274XX_UNSEAL_KEY);
+	if (status < 0) {
+		LOG_ERR("Unable to unseal the battery");
+		return -EIO;
+	}
+
+	status = bq274xx_control_reg_write(data, BQ274XX_UNSEAL_KEY);
+	if (status < 0) {
+		LOG_ERR("Unable to unseal the battery");
+		return -EIO;
+	}
+
+	status = bq274xx_control_reg_write(data,
+					   BQ274XX_CONTROL_SHUTDOWN_ENABLE);
+	if (status < 0) {
+		LOG_ERR("Unable to enable shutdown mode");
+		return -EIO;
+	}
+
+	status = bq274xx_control_reg_write(data, BQ274XX_CONTROL_SHUTDOWN);
+	if (status < 0) {
+		LOG_ERR("Unable to enter shutdown mode");
+		return -EIO;
+	}
+
+	status = bq274xx_control_reg_write(data, BQ274XX_CONTROL_SEALED);
+	if (status < 0) {
+		LOG_ERR("Failed to seal the gauge");
+		return -EIO;
+	}
+
+	data->pm_state = DEVICE_PM_OFF_STATE;
+
 	return 0;
 }
 
+static int bq274xx_exit_shutdown_mode(const struct device *dev)
+{
+	struct bq274xx_data *bq274xx = dev->data;
+	const struct bq274xx_config *const config = dev->config;
+	int status = 0;
+
+	bq274xx->gpio = device_get_binding((char *)config->gpio_port);
+	if (!bq274xx->gpio) {
+		LOG_DBG("Gpio controller %s not found.", config->gpio_port);
+		return -EINVAL;
+	}
+
+	status = gpio_pin_configure(bq274xx->gpio, config->int_pin,
+			   GPIO_OUTPUT | GPIO_OPEN_DRAIN);
+	if (status < 0) {
+		LOG_ERR("Unable to configure interrupt pin to output and open drain");
+		return status;
+	}
+
+	status = gpio_pin_set(bq274xx->gpio, config->int_pin, 0);
```

Comment:
@avisconti I can't seem to find the `*_dt` equivalent of `gpio_pin_set` function. I assume I'll need to set the pin how it is right now, with `gpio_pin_set`?

Reply:
Yes, correct. You can leave gpio_pin_set() as it is.